ocaml/xapi/xapi_pool_helpers.ml

```diff
@@ -61,6 +61,7 @@ let valid_operations ~__context record _ref' =
     | [_] -> set_errors Api_errors.cluster_already_exists [] [ `cluster_create ]
     (* indicates a bug or a need to update this code (if we ever support multiple clusters in the pool *)
     | _::_ -> failwith "Multiple clusters exist in the pool"
+    | [] -> if ha_enabled then raise (Api_errors.Server_error(Api_errors.incompatible_cluster_stack_active, ["xhad"]))
```

Comment:
You might want to get the actual cluster stack name from the pool instead of hardcoding it: `let current_stack = Db.Pool.get_ha_cluster_stack ~__context ~self:pool in`